allensdk/brain_observatory/filtered_running_speed/filtered_running_speed.py

```diff
@@ -0,0 +1,382 @@
+
+        return (
+            behavior_start,
+            behavior_end,
+            mapping_start,
+            mapping_end,
+            replay_start,
+            replay_end
+        )
+
+    def _merge_dx_data(
+        self,
+        mapping_velocities: pd.core.frame.DataFrame,
+        behavior_velocities: pd.core.frame.DataFrame,
+        replay_velocities: pd.core.frame.DataFrame,
+        frame_times: np.ndarray
+    ) -> list:
+        """
+        Concatenate all of the running speed data
+
+        Parameters
+        ----------
+        mapping_velocities: pandas.core.frame.DataFrame
+            Velocity data from mapping stimulus
+        behavior_velocities: pandas.core.frame.DataFrame
+           Velocity data from behavior stimulus
+        replay_velocities: pandas.core.frame.DataFrame
+            Velocity data from replay stimulus
+        frame_times: numpy.ndarray
+            list of the vsync times
+
+        Returns
+        -------
+        list[pd.DataFrame, pd.DataFrame]
+            concatenated velocity data, raw data
+        """
+
+        speed = np.concatenate(
+            (
+                behavior_velocities['speed'],
+                mapping_velocities['speed'],
+                replay_velocities['speed']),
+            axis=None
+        )
+
+        dx = np.concatenate(
+            (
+                behavior_velocities['dx'],
+                mapping_velocities['dx'],
+                replay_velocities['dx']),
+            axis=None
+        )
+
+        vsig = np.concatenate(
+            (
+                behavior_velocities['v_sig'],
+                mapping_velocities['v_sig'],
+                replay_velocities['v_sig']),
+            axis=None
+        )
+
+        vin = np.concatenate(
+            (
+                behavior_velocities['v_in'],
+                mapping_velocities['v_in'],
+                replay_velocities['v_in']),
+            axis=None
+        )
+
+        frame_indexes = list(
+            range(FilteredRunningSpeed.START_FRAME, len(frame_times))
+        )
+
+        velocities = pd.DataFrame(
+            {
+                "velocity": speed,
+                "net_rotation": dx,
+                "frame_indexes": frame_indexes,
+                "frame_time": frame_times
+            }
+        )
+
+        # Warning - the 'iscose' line below needs to be refactored
+        # is it exists in multiple places
+
+        # due to an acquisition bug (the buffer of raw orientations
+        # may be updated more slowly than it is read, leading to
+        # a 0 value for the change in orientation over an interval)
+        # there may be exact zeros in the velocity.
+        velocities = velocities[~(np.isclose(velocities["net_rotation"], 0.0))]
+
+        raw_data = pd.DataFrame(
+            {"vsig": vsig, "vin": vin, "frame_time": frame_times, "dx": dx}
+        )
+
+        return velocities, raw_data
+
+    def _process_single_simulus_experiment(
```

Comment:
Should this say "stimulus"? It looks like that typo propagated throughout this file (unless "simulus" is a word I don't know yet)

Reply:
Yes. Sorry. Thank you for pointing this out. I'm very dyslexic.